ktesting/testinglogger.go

```diff
@@ -61,14 +89,147 @@ func NewLogger(t TL, c *Config) logr.Logger {
 		prefix: "",
 		values: nil,
 		config: c,
+		buffer: new(buffer),
 	})
 }
 
+// Buffer stores log entries as formatted text and structured data.
+// It is safe to use this concurrently.
+//
+// Experimental
+//
+// Notice: This interface is EXPERIMENTAL and may be changed or removed in a
+// later release.
+type Buffer interface {
+	// String returns the log entries in the klog text
+	// output. The logger options determine whether
+	// this includes the header with time stamps.
+	String() string
+
+	// Data returns the log entries as structs.
+	Data() Log
+}
+
+// Log contains log entries in the order in which they were generated.
+//
+// Experimental
+//
+// Notice: This type is EXPERIMENTAL and may be changed or removed in a
+// later release.
+type Log []LogEntry
+
+// DeepCopy returns a copy of the log. The error instance and key/value
+// pairs remain shared.
+//
+// Experimental
+//
+// Notice: This function is EXPERIMENTAL and may be changed or removed in a
+// later release.
+func (l Log) DeepCopy() Log {
+	log := make(Log, 0, len(l))
+	for _, entry := range l {
+		log = append(log, entry)
+	}
```

Comment:
`log.append` doesn't work because append is only a stand-alone function, but avoiding the for loop is a good suggestion ->

`log = append(log, l...)`

Reply:
> 	logs = logs.append(l...)

Oh 🤦  Side effect of not having code auto completion.